tests/integration-tests/tests/proxy/test_proxy.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+from cfn_stacks_factory import CfnStack
+from remote_command_executor import RemoteCommandExecutor
+from utils import generate_stack_name
+
+from tests.common.schedulers_common import SlurmCommands
+
+
+@pytest.fixture(scope="class")
+def proxy_stack_factory(region, request, cfn_stacks_factory):
+    """
+    Set up and tear down a CloudFormation stack to deploy a proxy environment.
+
+    The stack is created from a CloudFormation template that sets up the necessary
+    networking, instances, and security groups for a proxy environment. The proxy
+    address and subnet are retrieved from the stack outputs and used in the cluster
+    configuration.
+    """
+    proxy_stack_template_path = "../../cloudformation/proxy/proxy.yaml"
+    with open(proxy_stack_template_path) as proxy_stack_template:
+        if request.config.getoption("proxy_stack"):
```

Comment:
thank you for adding this option! We should document the new option in test README